### PR TITLE
Implement non-opus multiple frames per callback solution

### PR DESCRIPTION
**Pull request by adambailey-**
This is basically a workaround to me not being able to correctly use the opus repacketizer in a reasonable amount of time. This is not a good long-term solution, but it functionally works (I see around 50% CPU or so when recording), which, while not perfect, is a step in the right direction.

### CODE REVIEW

File: src/encoderWorker.js [hunk 1]
```diff
@@ -70,7 +70,9 @@ var OggOpusEncoder = function( config, Module ){
     numberOfChannels: 1,
     originalSampleRate: 44100,
     resampleQuality: 3, // Value between 0 and 10 inclusive. 10 being highest quality.
-    serial: Math.floor(Math.random() * 4294967296)
+    serial: Math.floor(Math.random() * 4294967296),
+    streamOpusPackets: true,
+    framesPerCallback: 1 // Value between 1 and 2 inclusive. Only used when streamOpusPackets is true.
```
Comment (vocoded) [inline src/encoderWorker.js]: is it possible to allow this to be higher?  if not, then i'd make it a boolean value (`cacheFrameForCallback`), but if so then i'd generalize the code below to use the config value and return a variable length array

File: src/encoderWorker.js [hunk 2]
```diff
@@ -329,8 +332,20 @@ OggOpusEncoder.prototype.segmentPacket = function( packetLength ) {
...
+      } else if ( ++this.framesInCallback === this.config.framesPerCallback ) {
+        var cachedSegmentLength = this.segmentTable[ this.segmentTableIndex - 2 ];
+        var cachedSegment = this.segmentData.subarray( this.segmentDataIndex - cachedSegmentLength, this.segmentDataIndex );
+
+        global['postMessage']({ type: 'opus', data: [ cachedSegment, segment ] });
+        this.framesInCallback = 0;
+      }
+    }
+
     this.segmentData.set( segment, this.segmentDataIndex );
```
Comment (vocoded) [inline src/encoderWorker.js:348]: does this modify `segment`?

File: src/encoderWorker.js [hunk 3]
```diff
@@ -70,7 +70,9 @@ var OggOpusEncoder = function( config, Module ){
     numberOfChannels: 1,
     originalSampleRate: 44100,
     resampleQuality: 3, // Value between 0 and 10 inclusive. 10 being highest quality.
-    serial: Math.floor(Math.random() * 4294967296)
+    serial: Math.floor(Math.random() * 4294967296),
+    streamOpusPackets: true,
```
Comment (vocoded) [inline src/encoderWorker.js:74]: since this is in the `OggOpusEncoder` class, we could just call this `streamPackets`

**Review by vocoded — APPROVED** [review]
Minor suggestions; didn't test but assume it works

**Review by megamk — APPROVED** [review]

File: src/encoderWorker.js (hunk 1 above)
Comment (adambailey-) [reply src/encoderWorker.js]: I'll make it a boolean. Making it variable would be slightly harder, and not actually used by us, so I thought it wasn't worth it. If we ever made a PR back to the original repository, I'd want to have the Opus solution anyway.

File: src/encoderWorker.js (hunk 3 above)
Comment (adambailey-) [reply src/encoderWorker.js:74]: I think that would be confused with "streaming ogg opus packets", which we aren't doing

File: src/encoderWorker.js (hunk 2 above)
Comment (adambailey-) [reply src/encoderWorker.js:348]: Nope! It just takes the `segment` data and adds it to the `segmentData` array at an offset (`this.segmentDataIndex`)

**Pull request merged**